### ISSUE TITLE
Remove exif metadata from all the embedded images that are inside a PDF document

### ISSUE THREAD

**Issue by Geo-Van** [qpdf/qpdf#1168]
The following command it is well known that it FLATTENS ANNOTATIONS & REMOVE METADATA from a PDF file:
qpdf --flatten-annotations=all --empty --pages input.pdf -- output.pdf

BUT, please let me ask:
Is there a way  to remove exif metadata from all the embedded images that are inside a PDF document, please?

**Comment by jberkenbilt** [comment]
No, sorry, there are no qpdf commands that touch exif metadata in embedded images. It would be possible to write something that would go through, detect image streams with /DCTDecode filtering, and looking at the streams to see if they are JFIF files. If so, the stream could be replaced by a new stream that removes the Exif metadata. Something may be similarly possible for JBIG2 or JP2 data. For all other image types, there will be no embedded Exif metadata. Doing this would be out of scope for the qpdf library or CLI though and will not be added as a feature.

**Comment by Geo-Van** [comment]
Thank you very much for let me know this.

**Comment by rubyFeedback** [comment]
Geo-Van wrote:

> Is there a way to remove exif metadata from all the embedded images that are inside a PDF document, please?

I think you can probably do this with ImageMagick?

I haven't tried that yet, but I think ImageMagick could be used, and perhaps qpdf could be used to extract the images, then have imagemagick iterate over them, then "re-assemble" the new images back into the .pdf file via qpdf. At the least this is my thought process; for the batch job you could use ruby or python or just a shell script probably. I haven't researched this yet, though - usually I use imagemagick to e. g. query width and height of images, and / or setting the title to the filename of the .pdf file at hand and so forth.

I guess Jay understandably focuses on a subset of things for qpdf to do (and do well); more and more code also means more things to maintain, more potential bugs, more documentation to be written and so forth.
